### PR TITLE
Upgrade eslint configuration and fix throw literal warnings

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
None

#### What's this PR do?
Upgrades eslint configuration, fixes related warnings

#### How should this be manually tested?
N/A

### CODE REVIEW

File: static/js/dashboard_routes.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import Learner from './components/Learner';
 import OrderSummaryPage from './containers/OrderSummaryPage';
 
 const errorLoading = error => {
-  console.trace(error);
+  console.log("loading error caused by:", error)
```
Comment (amir-qayyum-khan) [inline static/js/dashboard_routes.js]: why console here?

Comment (noisecapella) [reply static/js/dashboard_routes.js]: `console.trace` was showing the stack trace of `errorLoading`, which doesn't contain any useful information. Instead we want to see the stack trace of `error` which shows up via `console.log`

**Comment by amir-qayyum-khan** [comment]
@noisecapella just out of curiosity, You choosed `thorw new Error` which logs `error , (custom message)` to be more readable?

**Comment by noisecapella** [comment]
No, in javascript the stack trace of an `Error` object is defined at the point when the `Error` was created, not when it was thrown. In order to get the stack trace of a caught error it needs to be an `Error` object.

**Pull request merged**